tests/TestCase/TestSuite/TestCaseTest.php

```diff
@@ -146,6 +146,48 @@ public function testWithErrorReportingWithException(): void
         }
     }
 
+    /**
+     * test deprecated
+     */
+    public function testDeprecated(): void
+    {
+        $this->deprecated(function () {
+            trigger_error('deprecation message', E_USER_DEPRECATED);
```

Comment:
You could make different messages to get around the duplicate prevention 🤷

Reply:
this commit was before my last commit

Reply:
and we just need a deprecation warning not complex `deprecationwarning()`, and this helps us to maintaining easier
if needed I can put a  `deprecationwarning()`

Reply:
I made this suggestion when I thought the removal of the duplicate detection was because it made writing these tests easier. However, that was before I knew about the additional use-cases you outlined in https://github.com/cakephp/cakephp/pull/15935#discussion_r716800070. I no longer think anything needs to change from the current diff.